Source/Core/VideoBackends/OGL/BoundingBox.cpp

```diff
@@ -2,69 +2,160 @@
 // Licensed under GPLv2+
 // Refer to the license.txt file included.
 
+#include <algorithm>
+#include <array>
 #include <cstring>
 
 #include "Common/GL/GLUtil.h"
 
 #include "VideoBackends/OGL/BoundingBox.h"
+#include "VideoBackends/OGL/FramebufferManager.h"
 
 #include "VideoCommon/DriverDetails.h"
 #include "VideoCommon/VideoConfig.h"
 
 static GLuint s_bbox_buffer_id;
 
+static std::array<int, 4> s_stencil_bounds{{0, 0, 0, 0}};
+static std::unique_ptr<u8[]> s_stencil_data;
+static bool s_stencil_updated;
+static bool s_stencil_cleared;
+
+static int s_target_width;
+static int s_target_height;
+
 namespace OGL
 {
-void BoundingBox::Init()
+void BoundingBox::SetEFBScaleChanged(float xscale, float yscale)
+{
+  if (g_ActiveConfig.backend_info.bSupportsFragmentStoresAndAtomics)
+    return;
+
+  s_target_width = static_cast<int>(EFB_WIDTH * xscale);
```

Comment:
Sorry if I'm missing something here, but is there a reason that you are recalculating the width/height here? Why not just read the target width/height from Renderer?